File: src/main/java/com/github/xumeng/java/problems/easy/problem0338/Solution0338.java
package com.github.xumeng.java.problems.easy.problem0338;

public class Solution0338 {

    public int[] countBits(int n) {
        int arrayLength = n + 1;
        int[] result = new int[arrayLength];
        for (int i = 0; i < arrayLength; i++) {
            result[i] = countNumberOfOne(i);
        }
        return result;
    }

    private int countNumberOfOne(int n) {
        int quotient;
        int countOfOne = 0;
        while (n != 0) {
            quotient = n / 2;
            if (n % 2 == 1) {
                countOfOne++;
            }
            n = quotient;
        }
        return countOfOne;
    }

}

File: src/main/kotlin/com/github/xumeng/kotlin/problems/easy/problem0338/Solution0338.kt
package com.github.xumeng.kotlin.problems.easy.problem0338

class Solution0338 {

    fun countBits(n: Int): IntArray {
        val result = IntArray(n + 1)
        for (i in 0..n) {
            result[i] = countNumberOfOne(i)
        }
        return result
    }

    private fun countNumberOfOne(n: Int): Int {
        var dividend: Int = n
        var quotient: Int
        var countOfOne = 0
        while (dividend != 0) {
            quotient = dividend / 2
            if (dividend % 2 == 1) {
                countOfOne++
            }
            dividend = quotient
        }
        return countOfOne
    }

}